launch_xml/launch_xml/entity.py

```diff
@@ -0,0 +1,117 @@
+# limitations under the License.
+
+"""Module for Entity class."""
+
+from typing import List
+from typing import Optional
+from typing import Text
+from typing import Type
+from typing import Union
+import xml.etree.ElementTree as ET
+
+from launch.launch_frontend import Entity as BaseEntity
+from launch.launch_frontend.type_utils import get_typed_value
+from launch.launch_frontend.type_utils import SomeAllowedTypes
+
+
+class Entity(BaseEntity):
+    """Single item in the intermediate XML front_end representation."""
+
+    def __init__(
+        self,
+        xml_element: ET.Element = None,
+        *,
+        parent: 'Entity' = None
+    ) -> Text:
+        """Construnctor."""
+        self.__xml_element = xml_element
+        self.__parent = parent
+
+    @property
+    def type_name(self) -> Text:
+        """Get Entity type."""
+        return self.__xml_element.tag
+
+    @property
+    def parent(self) -> Optional['Entity']:
+        """Get Entity parent."""
+        return self.__parent
+
+    @property
+    def children(self) -> List['Entity']:
+        """Get the Entity's children."""
+        return [Entity(item) for item in self.__xml_element]
+
+    def get_attr(
+        self,
+        name: Text,
+        *,
+        types:
+            Optional[
+                Union[
+                    SomeAllowedTypes,
+                    Type[List[BaseEntity]],
+                    Type[List['Entity']],
+                ]
+            ] = str,
+        optional: bool = False
+    ) -> Optional[Union[
+        Text,
+        int,
+        float,
+        List[Text],
+        List[int],
+        List[float],
+        List['Entity']
+    ]]:
+        """Access an attribute of the entity."""
+        attr_error = AttributeError(
+            'Attribute {} of type {} not found in Entity {}'.format(
+                name, types, self.type_name
+            )
+        )
+        is_list_entity = types is not None and not isinstance(types, tuple) \
+            and issubclass(types, List) and issubclass(types.__args__[0], BaseEntity)
+        if is_list_entity:
+            return_list = filter(lambda x: x.tag == name, self.__xml_element)
+            if not return_list:
+                if optional:
+                    return None
+                else:
+                    raise attr_error
+            return [Entity(item) for item in return_list]
+        value = None
+        if name in self.__xml_element.attrib:
+            name_sep = name + '-sep'
+            if name_sep not in self.__xml_element.attrib:
+                value = self.__xml_element.attrib[name]
+            else:
+                sep = self.__xml_element.attrib[name_sep]
+                value = self.__xml_element.attrib[name].split(sep)
+        if value is None:
+            if not optional:
+                raise attr_error
+            else:
+                return None
+        try:
+            value = get_typed_value(value, types)
```

Comment:
@ivanpauno hmm why are we mixing YAML and XML here?

Reply:
I'm only doing the conversions from string to int/float/bool like yaml, and allowing the yaml ways of escaping them. If not it was imposible to specify an string param of value "1" (now you can do that passing "'1'".

Maybe this is not the best solution, if you have any other ideas please tell me.

Maybe, one option is to add an optional type field in `param`. If you don't use it, the automatic type conversions are done as before ("1" is an int "'1'" doesn't escape the quotes, you see them).
If you want a string param containing 1 you could do:
```xml
<param name="asd" value="1" type="str"/>
```

I like this last option. It should be possible to use the `type` attribute in yaml too, but it hasn't much sense for the user.

Reply:
My point is that with `get_typed_value()`'s current implementation, you can pass a list of floats using `[1, 2, 3]`. And then we also have value separators attributes.

Reply:
I restricted that conversion in https://github.com/ros2/launch/pull/226/commits/18cf4a06acaa0b5002e7b5c5f4fe90e03b08651d.
For parameters inside substitutions, you can still pass a list like yaml (becuase of https://github.com/ros2/launch_ros/pull/31).
So this example is still working:
https://github.com/ros2/launch_ros/blob/713fe0034d89c412dd0efbd327ce31aecdb359c0/test_launch_ros/test/test_launch_ros/launch_frontend/test_node_frontend.py#L31L50